src/components/menu/EnterButton.tsx

```diff
@@ -1,50 +1,21 @@
 import React from "react";
-import workLogger from "../../api/workLogger";
 import "./EnterButton.scss";
-import { ButtonActionProps } from "../../types/types";
 import Button from "../basics/Button";
 import Container from "../basics/Container";
 
-const EnterButton = (props: ButtonActionProps) => {
+export type EnterButtonProps = {
+    inOffice: boolean;
+    onClick: () => void;
+};
 
-    const handleClick = async () => {
-        try {
-            if(!props.inOffice) {
-                await workLogger.post("/log", {
-                    dateString: new Date().toString(),
-                    userEmail: props.userEmail
-                });
-                props.trackLogRequest({ success: true, inOffice: true });
-            }
-        } catch(err) {
-            if(err.message === "Network Error") {
-                props.trackLogRequest({ success: true, inOffice: true });
-            } else {
-                props.trackLogRequest({ success: false, inOffice: false, bannerMessage: "Something Went Wrong!" });
-            }
-        }
-    }
```

Comment:
These function we're pretty hard to understand, and kind of repeating themselves.
Also this logic really belongs higher up in the chain